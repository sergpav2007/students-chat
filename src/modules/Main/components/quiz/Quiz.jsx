import React from 'react';
import { QuizWrapper } from './styledComponents';
import Game from './components/game';
import Results from './components/results';
import ReadyForGame from './components/readyForGame/ReadyForGame';

const Quiz = ({
    isShowResults,
    isQuizInProcess,
    setIsReadyForGame,
    isUserReadyToStartQuiz,
}) => {
    return (
        <QuizWrapper>
            {isQuizInProcess ?
                !isShowResults ?
                    <Game/>
                    :
                    <Results/>
                :
                <ReadyForGame
                    callback={setIsReadyForGame}
                    isUserReadyToStartQuiz={isUserReadyToStartQuiz}
                />
            }
        </QuizWrapper>
    );
};

export default Quiz;
