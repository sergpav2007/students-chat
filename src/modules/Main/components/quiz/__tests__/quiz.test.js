import Quiz from '../Quiz';

describe('<Quiz/>', () => {
    let wrapper;

    const props = {
        isShowResults: false,
        isQuizInProcess: true,
        setIsReadyForGame: jest.fn(() => {}),
        isUserReadyToStartQuiz: true,
    };

    beforeEach(() => {
        wrapper = shallow(<Quiz {...props}/>);
    });

    it('snapshot created, should rendered correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Render correctly with snapshot (renderer.create)', () => {
        mountSmart(<Quiz {...props}/>, store);
    });
});
