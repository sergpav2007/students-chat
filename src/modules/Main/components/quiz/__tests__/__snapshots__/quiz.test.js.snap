// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quiz/> snapshot created, should rendered correctly 1`] = `
<styled.div>
  <Connect(Game) />
</styled.div>
`;
